sale_coupon_limit/models/sale_coupon_rule.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2021 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class SaleCouponRule(models.Model):
+    _inherit = "sale.coupon.rule"
+
+    rule_max_customer_application = fields.Integer(
+        string="Maximum Customer Applications",
+        default=0,
+        help="Maximum times that a program can be applied to a customer. "
+        "0 for no limit.",
+    )
+    rule_salesmen_limit_ids = fields.One2many(
+        string="Salesmen Limits",
+        comodel_name="sale.coupon.rule.salesmen.limit",
+        inverse_name="rule_id",
+        help="Maximum times a salesman can apply a program. 0 for no limit.",
+    )
+    rule_salesmen_restrict_list_only = fields.Boolean(
+        string="Restrict on listed salesmen",
+        default=True,
```

Comment:
Precisely the default is the standard behavior. So when is checked, the rules apply to the list only, otherwise they apply to the list and everyone else is banned.

Reply:
If you restrict on listed salesmen by default, and no salesmen is put, then you are always forbidding AFAIK, and we don't want that.

Reply:
Well, it's a naming issue maybe :) The field restricts the salesmen limits, to the list (the rest of users have no limits, as they don't have restrictions). When check, the limits apply to everyone, an a user not in the list can't use that promotion.

Reply:
Please change the name and feature then for making it more comprehensive:
```
rule_salesmen_strict_limit = fields.Boolean(default=False, string="Strict limit", help="If marked, promotion will only allowed for the list of salesmen with their quantities")
```

Reply:
done